apis/v1alpha2/httproute_types.go

```diff
@@ -598,6 +614,24 @@ type HTTPRequestHeaderFilter struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
+// HTTPRequestRedirect defines configuration for the RequestRedirect filter.
+type HTTPRequestRedirect struct {
+	// AutoHTTPS if set to true redirects any incoming HTTP requests to HTTPS.
+	// The path and querystring of the request MUST be preserved in the Location
+	// header in the response.
+	AutoHTTPS bool `json:"auto_https,omitempty"`
+	// TODO if defaulting is possible here (not sure about defaulting in filter), default this?
+	// What status codes should be allowed? 301,302,307,308?
```

Comment:
If we have bounded ones one option we should consider is status code number vs an enum? like MOVED_PERMANENTLY, etc.

Reply:
My concern with enum is it is one more piece of configuration that the user needs to learn and remember.
We could  prevent errors via validation here if that is the concern.

Reply:
Agree that I think numbers will likely be easier to understand here.